src/Admin/Controller/SendTestMailController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Admin\Controller;
+
+use Flarum\Foundation\Application;
+use Flarum\Settings\TemporarySettingsRepository;
+use Flarum\User\AssertPermissionTrait;
+use Illuminate\Contracts\Validation\Factory;
+use Illuminate\Support\Arr;
+use Laminas\Diactoros\Response\JsonResponse;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Swift_Mailer;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class SendTestMailController implements RequestHandlerInterface
+{
+    use AssertPermissionTrait;
+
+    protected $app;
+
+    protected $translator;
+
+    public function __construct(Application $app, TranslatorInterface $translator)
```

Comment:
Please typehint the `Container` contract instead. You're only using the container methods, as far as I can tell, so that's a bit more decoupled.

(You'll receive the same instance in the end, but this class doesn't need to know that. :wink:)

Reply:
Done